### ISSUE TITLE
Evaluate option to use CRDs as storage for approvals and any other things

### ISSUE THREAD

**Issue by rusenask** [keel-hq/keel#516]
It might make sense to use CRDs to store approvals instead of sqlite, less state for the pod and no need for a volume.

**Comment by mavimo** [comment]
Maybe CRD should be used to keep track of images and a configmap / secret should be enough for approva if we don't want to force users to install CRDs

**Comment by rusenask** [comment]
yeah, I was thinking about it too, crd installation is a bit heavy-weight compared to anything else, however it's not unexpected :) If we do go with CRD though, it would allow us to expand functionality a bit and store config in a more structure way (as you say for example for images, potentially credentials for registries, etc)
